cypress/integration/user_roles.js

```diff
@@ -0,0 +1,46 @@
+import LoginPage from '../objects/loginPage'
+import TimeLoggingPage from '../objects/timeLoggingPage';
+
+describe('Sourcebooks login', function() {
+
+    const loginPage = new LoginPage()  
+    const timeLogging = new TimeLoggingPage();    
+
+    it('Should display validation for empty user after attempted loggin', function () {
+        
+
+        loginPage.visit()
+
+        loginPage.getValidationError().should('not.visible')
+        loginPage.submit().click();
+        loginPage.getValidationError().should('be.visible')
+    })
+
+    const roles= [['User', 1], ["Team Lead", 2], ['Manager', 5], ['Accountant', 5], ['Admin', 6]]
+
+    for (let i = 0; i < roles.length; i++) {
+
+        it('Should be able to login with each role User', function () {
```

Comment:
It would be great to change the name of this test case to user elements from the "roles" array as the parameter and also remove the word "each", e.g. 

``Should be able to login with role ${roles[i][0]}``

This way every test case iteration will have the appropriate role name :)